kornia/augmentation/_2d/intensity/random_snow.py

```diff
@@ -0,0 +1,65 @@
+from typing import Any, Dict, Optional
+
+from kornia.augmentation import random_generator as rg
+from kornia.augmentation._2d.intensity.base import IntensityAugmentationBase2D
+from kornia.color import hls_to_rgb, rgb_to_hls
+from kornia.core import Tensor
+from kornia.core.check import KORNIA_CHECK
+
+
+class RandomSnow(IntensityAugmentationBase2D):
+    r"""Generates snow effect on given tensor image or a batch tensor images.
+
+    Args:
+        snow_coefficient: A float between 0 and 1 that controls the amount of snow to add to the
+        image, the larger value corresponds to the more snow.
+        brightness: A float greater than 1 that controls the brightness of the snow.
+        same_on_batch: If True, apply the same transformation to each image in a batch. Default: False.
+        p: Probability of applying the transformation. Default: 0.5.
+        keepdim: Keep the output tensor with the same shape as input. Default: False.
+
+    Shape:
+        - Input: :math:`(C, H, W)` or :math:`(B, C, H, W)`
+        - Output: :math:`(B, C, H, W)`
+
+    Examples:
+        >>> inputs = torch.rand(2, 3, 4, 4)
+        >>> snow = kornia.augmentation.RandomSnow(p=1.0, snow_coefficient=0.2, brightness=2.0)
+        >>> output = snow(inputs)
+        >>> output.shape
+        torch.Size([2, 3, 4, 4])
+    """
+
+    def __init__(
+        self,
+        snow_coefficient: float = 0.5,
+        brightness: float = 2.5,
+        same_on_batch: bool = False,
+        p: float = 0.5,
+        keepdim: bool = False,
+    ) -> None:
+        super().__init__(p=p, same_on_batch=same_on_batch, keepdim=keepdim)
+        KORNIA_CHECK(0 <= snow_coefficient <= 1, "Snow coefficient must be between 0 and 1.")
+
+        self._param_generator = rg.PlainUniformGenerator(
+            ((snow_coefficient, snow_coefficient), "snow_coefficient", 0.5, (0.0, 1.0)),
+            ((brightness, brightness), "brightness", None, None),
+        )
```

Comment:
Is it preferred to fix snow and brightness for this augmentation? Otherwise, directly making those paramters as tuples shall be better.

Reply:
If I understand correctly, if they are fixed, the augmentation won't be random, as you mentioned earlier. I thought this piece of code randomize these parameters. What do we need to change?

Reply:
You need to alter the lower and upper bound. You currently set them as the same, so that no randomness will happen even if the code supports it.

I suggest to make the tuple arguments in __init__ and let the user decide the lower and upper bounds.